### PR TITLE
Add support for loading custom embeddings.

### PR DESCRIPTION
**Pull request by nsthorat**
- Add `Dataset.load_embeddings`, which takes a lambda, an index_path (the text field) and loads embeddings into our vector store.
- Add `ll.register_embedding` which wraps `ll.register_signal`
- Rename `ll.lilac_embedding` => `ll.chunk_embedding` which is now public API.
- Add `Dataset.delete_embedding` which deletes an embedding and its index.

Fixes https://github.com/lilacai/lilac/issues/1080 and https://github.com/lilacai/lilac/issues/1087

Documentation:
- Ad of docs for compute custom embeddings, and loading embeddings.
- Add an accompanying notebook for loading custom embeddings.

Other fixes:
- Centralize the get_vector_index since it was used in two places
- Add exists_ok to register_signal in case of double registration

### CODE REVIEW

**Review by dsmilkov — APPROVED** [review]
awesome!

File: lilac/data/dataset_duckdb.py [hunk 1]
```diff
@@ -1221,6 +1235,161 @@ def compute_embedding(
...
+
+    # Delete the signal manifest.
+    signal_manifest_filepath = os.path.join(output_dir, SIGNAL_MANIFEST_FILENAME)
+    # If the signal manifest already exists, delete it as it will be rewritten after the new signal
+    # outputs are run.
+    if os.path.exists(signal_manifest_filepath):
+      os.remove(signal_manifest_filepath)
+      # Call manifest() to recreate all the views, otherwise this could be stale and point to a non
+      # existent file.
+      self.manifest()
```
Comment (brilee) [inline lilac/data/dataset_duckdb.py:1265]: should call `self._clear_joint_table_cache()`, not self.manifest()

Comment (nsthorat) [reply lilac/data/dataset_duckdb.py]: Done

File: lilac/data/dataset_duckdb.py [hunk 2]
```diff
@@ -1221,6 +1235,161 @@ def compute_embedding(
...
+      if overwrite:
+        self.delete_embedding(embedding, index_path)
+      else:
+        raise ValueError(
+          f'Embedding "{embedding}" already exists at path {index_path}. '
+          'Use overwrite=True to overwrite.'
+        )
+
+    query_params = DuckDBQueryParams()
+    estimated_len = self.count(query_params)
```
Comment (brilee) [inline lilac/data/dataset_duckdb.py:1316]: you can set query_params default kwargs to None, then you can just call `self.count()`, which is the obvious thing to do.

**Review by brilee — APPROVED** [review]

File: lilac/data/dataset_duckdb.py (hunk 2 above)
Comment (nsthorat) [reply lilac/data/dataset_duckdb.py]: done

**Pull request merged**